app/controllers/sites_controller.rb

```diff
@@ -5,8 +5,33 @@ def index
     @sites = @provider.sites.sort_by(&:location_name)
   end
 
+  def show
+    redirect_to edit_provider_site_path
+  end
+
   def edit
-    @site = @provider.sites.find(params[:id]).first
+    @site = @provider.sites.select { |site| site.id == params[:id] }.first
+    @errors = {}
+  end
+
+  def update
+    sites = @provider.sites
+    @site = sites.select { |site| site.id == params[:id] }.first
+
+    @errors = {}
+    if params[:site] then
+      site_by_name = sites.select { |site| site.location_name == params[:site][:location_name] }.first
+      name_is_in_use = !site_by_name.nil? && site_by_name.id != @site.id
+      @errors[:location_name] = 'Enter a name that is not in use' if name_is_in_use
+
+      @errors[:location_name] = 'Enter a name' if params[:site][:location_name]&.empty?
+      @errors[:address1] = 'Enter an address' if params[:site][:address1]&.empty?
+      @errors[:address3] = 'Enter a town or city' if params[:site][:address3]&.empty?
+      @errors[:postcode] = 'Enter a valid postcode (for example, SW1P 3BT)' if params[:site][:postcode]&.empty?
```

Comment:
`SW1P 3BT` is a valid example but doesn't follow the standard postcode format (note the letter after the first number). Maybe use a more common postcode format like `BN1 1AA` instead?

Reply:
manage courses ui uses this message

https://github.com/DFE-Digital/manage-courses-ui/blob/875ae72320369dbec9528c245abb4b6274acac51/src/ui/ViewModels/OrganisationViewModel.cs#L64

Reply:
Done, good suggestion :+1: